From 08cea8140a56a2b889ad7ba5aef6ac1c3bb0a1a7 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Fri, 19 Jun 2009 09:44:38 +0800
Subject: [PATCH] ENGR00113478 IPUv1:Fix video play break without resizing and rotation

IPUv1 needs IC module to do CSC even if
there is no resizing and rotation requirement,
so we cannot bypass IC in this case.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 drivers/media/video/mxc/output/mxc_v4l2_output.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/output/mxc_v4l2_output.c b/drivers/media/video/mxc/output/mxc_v4l2_output.c
index 4c70599..1bb271d 100644
--- a/drivers/media/video/mxc/output/mxc_v4l2_output.c
+++ b/drivers/media/video/mxc/output/mxc_v4l2_output.c
@@ -907,6 +907,13 @@ static int mxc_v4l2out_streamon(vout_data * vout)
 			vout->ic_bypass = 0;
 		}
 
+#ifdef CONFIG_MXC_IPU_V1
+		/* IPUv1 needs IC to do CSC */
+		if (format_is_yuv(vout->v2f.fmt.pix.pixelformat) !=
+		    format_is_yuv(bpp_to_fmt(fbi)))
+			vout->ic_bypass = 0;
+#endif
+
 		fbvar = fbi->var;
 
 		if (vout->cur_disp_output == 3) {
-- 
1.5.4.4

